[tests] fix Java.Interop test results copying

jonathanpeppers opened the pull request:
On Windows, under MSBuild, I noticed a clean build was not copying the
test results from `external/Java.Interop` to the root directory. Two
things were causing this to be overlooked: the `<Copy />` task left the
original files from past test runs, and the `@(_JavaInteropTestResults)`
`ItemGroup` was calculated at the top of the file. I *think* this is
somehow working on xbuild.

To fix this, I moved the location of the `@(_JavaInteropTestResults)`
`ItemGroup` to happen during the target, instead of at the top of the
file. I also added a `<Delete />` task after the `<Copy />` to keep a clean
`external/Java.Interop` directory. It doesn't look like xbuild supports
`<Move />`.

dnfclas commented:
@jonathanpeppers,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

jonathanpeppers commented:
I need to bump Java.Interop for https://github.com/xamarin/java.interop/pull/209

jonpryor commented:
Is this PR moot since PR #1022 has been merged?

jonathanpeppers commented:
@jonpryor I removed a commit and rebased, there still seems to be a problem on Windows: https://devdiv.visualstudio.com/DevDiv/Default/_build/index?buildId=1131732&_a=summary&tab=ms.vss-test-web.test-result-details

The TestResults files aren't getting moved to the root of the repo.
